### PR TITLE
ID22563: Error handling

### PR DESCRIPTION
**Pull request by sviridovartem**

### CODE REVIEW

File: webapplication/sip/settings.py [hunk 1]
```diff
@@ -251,4 +251,6 @@
 SITE_ID = 1
 
 DEFAULT_SYSTEM_NOTIFICATION_EMAIL = ""
+DEFAULT_USER_TRANSACTION_STATUS = "Something went wrong, please contact us"
```
Comment (adolgarev) [inline webapplication/sip/settings.py:254]: This is a bit strange for a default transaction status. The name is misleading.

File: webapplication/aoi/models.py [hunk 2]
```diff
@@ -119,10 +119,16 @@ class Request(models.Model):
...
     request_origin = models.CharField(max_length=50, default="https://portal.soilmate.ai/", verbose_name='Request origin')
 
     @property
     def component_name(self):
         return self.component.name
+
+
+class UserError(models.Model):
+    user_error = models.CharField(max_length=400, blank=True, null=True, verbose_name='User error')
+    component_error = models.CharField(max_length=400, blank=True, null=True, unique=True, verbose_name='Component error')
```
Comment (adolgarev) [inline webapplication/aoi/models.py:134]: The idea is good, although naming is not ok. Looking at the model I do not understand what model and it attiubutes represent.

File: webapplication/aoi/management/commands/_notebook.py [hunk 3]
```diff
@@ -23,6 +23,14 @@
 
 THREAD_SLEEP = 10
 
+def clean_container_logs(logs):
+    log_lines = logs.split('\n')
+    log_lines = [re.sub(r'^\s*\d+\s*', '', line) for line in log_lines]
+
+    log_text = ''.join(log_lines)
+    log_text = re.sub(r'\x1b\[[0-9;]*m', '', log_text)
+    return log_text
```
Comment (adolgarev) [inline webapplication/aoi/management/commands/_notebook.py:32]: An example of what it does would be nice.

File: webapplication/aoi/management/commands/_notebook.py [hunk 4]
```diff
@@ -164,20 +172,33 @@ def execute_notebook(self):
...
+                for error in known_errors:
+                    if error in request.error:
+                        request.user_error = UserError.objects.get(component_error=error).user_error
+                        request.save(update_fields=['user_error'])
+                        logger.info("Known error added")
+                        break
+                if not request.user_error:
+                    logger.info("No known error for component error")
                 request.save(update_fields=['error'])
+                #todo what if more then one error acceptable
```
Comment (adolgarev) [inline webapplication/aoi/management/commands/_notebook.py:191]: It should be a list.

File: webapplication/aoi/management/commands/_notebook.py [hunk 5]
```diff
@@ -164,20 +172,33 @@ def execute_notebook(self):
...
+                #todo remove trash from original error message
```
Comment (adolgarev) [inline webapplication/aoi/management/commands/_notebook.py:192]: Does clean_container_logs resolve this?

File: webapplication/user/models.py [hunk 6]
```diff
@@ -114,8 +114,12 @@ class Meta:
...
+    def generate_comment(request: Request, error=None):
         comment = f"{request.component_name}."
-        if aoi:
-            comment += f" Area: {aoi.name} processed."
+        if request.aoi:
+            comment += f" Area: {request.aoi.name}."
+        if not error:
+            comment += " Processing."
+        else:
+            comment += f" Error: '{error}'."
```
Comment (adolgarev) [inline webapplication/user/models.py:124]: I'd better (1) not materialize this at the transaction creation time, (2) move this logic to UI, transaction API should return necessary data about request such as its AOI (BTW it should always be set in request), corresponding errors (error ids).

File: webapplication/sip/settings.py (hunk 1 above)
Comment (sviridovartem) [reply webapplication/sip/settings.py]: fixed

File: webapplication/aoi/management/commands/_notebook.py (hunk 5 above)
Comment (sviridovartem) [reply webapplication/aoi/management/commands/_notebook.py:192]: yes, just forgot remove todo

File: webapplication/aoi/management/commands/_notebook.py (hunk 3 above)
Comment (sviridovartem) [reply webapplication/aoi/management/commands/_notebook.py]: fixed

File: webapplication/aoi/management/commands/_notebook.py (hunk 4 above)
Comment (sviridovartem) [reply webapplication/aoi/management/commands/_notebook.py]: fixed

File: webapplication/user/models.py (hunk 6 above)
Comment (sviridovartem) [reply webapplication/user/models.py]: partly fixed

**Pull request merged**